scipy/sparse/dok.py

```diff
@@ -497,30 +483,22 @@ def tocsc(self, copy=False):
     tocsc.__doc__ = spmatrix.tocsc.__doc__
 
     def resize(self, shape):
-        """ Resize the matrix in-place to dimensions given by 'shape'.
+        """Resize the matrix in-place to dimensions given by `shape`.
 
         Any non-zero elements that lie outside the new shape are removed.
         """
         if not isshape(shape):
-            raise TypeError("dimensions must be a 2-tuple of positive"
-                             " integers")
+            raise TypeError("Dimensions must be a 2-tuple of positive integers")
         newM, newN = shape
         M, N = self.shape
         if newM < M or newN < N:
             # Remove all elements outside new dimensions
-            for (i, j) in _list(self.keys()):
+            for (i, j) in list(self.keys()):
```

Comment:
Should be fixed, I also found another place which used `dict.keys()`. How would you go about measuring if a copy is made? I used `%memit` in ipython but did not find any memory difference for dictionary with 10^7 elements. But there is a measurable time difference, `dict.keys()` takes twice as much time compared to `dict.iterkeys()`.